pkg/cpu/cpu_darwin.go

```diff
@@ -0,0 +1,110 @@
+package cpu
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"os/exec"
+	"strconv"
+	"strings"
+)
+
+var sysctlOutput = make(map[string]string)
+
+func (i *Info) load() error {
+	err := populateSysctlOutput()
+	if err != nil {
+		return errors.Wrap(err, "unable to populate sysctl map")
+	}
+
+	i.TotalCores = getTotalCores()
+	i.TotalThreads = getTotalThreads()
+	i.Processors = getProcessors()
+
+	return nil
+}
+
+// getProcessors some more info https://developer.apple.com/documentation/kernel/1387446-sysctlbyname/determining_system_capabilities
+func getProcessors() []*Processor {
+	p := make([]*Processor, getProcTopCount())
+	for i, _ := range p {
+		p[i] = new(Processor)
+		p[i].Vendor = sysctlOutput[fmt.Sprintf("hw.perflevel%s.name", strconv.Itoa(i))]
+		p[i].Model = getVendor()
+		p[i].NumCores = getNumberCoresFromPerfLevel(i)
+		p[i].Capabilities = getCapabilities()
+	}
+	return p
+}
+
+// valid for ARM see https://developer.apple.com/documentation/kernel/1387446-sysctlbyname/determining_instruction_set_characteristics
+func getCapabilities() []string {
+	caps := []string{}
+	for k, v := range sysctlOutput {
+		if strings.HasPrefix(k, "hw.optional.arm.") {
```

Comment:
This should be guarded by a check that the system uses an ARM architecture. Consider putting this somewhere:

```go
var (
    hasARMArchitecture = false
)

// and inside your populateSysctlOutput function, do something like:
if k == "hw.optional.arm64" && v == "1" {
    hasARMArchitecture = true
}
```

If the host isn't an ARM architecture, none of the `hw.optional.arm` traits will be 1 and therefore there's no need to proceed.

Reply:
added the check